contributing-tips.rst

```diff
@@ -0,0 +1,142 @@
+
+The exception is when several issues are closely related or can reasonably be worked on together.  In this case, it should be clear by looking at the conversation on the Github issues that the items are related and will be worked on together.
+
+
+.. _descriptive-pr-names:
+
+Descriptive PR names
+----------------------
+
+A PR title should answer the question, "What does this Pull Request do?"
+
+Good PR titles:
+
+- adds a video directive
+- makes navigation buttons responsive
+- swaps placement of nav buttons and file an issue note
+
+Bad PR titles:
+
+- fix issue
+- fix #123
+- collect
+
+.. _small-commits:
+
+Small Atomic Commits
+-----------------------
+
+When working locally, commit often. Don't wait until you have 100 lines of changes across multiple files.
+
+- If you need to copy or move a large section or file, commit that change before also editing it. 
+- If you have to create a new template file based on an existing template file, copy the file in one commit and then work on the changes. This makes it easier to know what you actually did.
+- If writing a new doc, commits after each new section are a good idea.
+
+Commit messages should answer the question, "What does this commit do?"
+
+Small, well-named commits will help you keep track of your own work and make rollbacks and other changes easier to deal with.
+
+.. _discuss-issues:
+
+Discuss issues before working
+--------------------------------
+
+Take the time to clarify the needs and scope of an issue before committing to work on it. Especially for coding tasks, make sure you state your understanding and your plan before working. 
+
+If you have a question, ask. Don't guess.
+
+.. note::
+
+  Many new contributors don't ask questions because they are worried about appearing under informed. Please set this worry aside.
+
+  **You will never be judged harshly for asking clarifying questions or for seeking more information.** 
+
+.. _if-you-get-stuck:
+
+If you get stuck while working
+--------------------------------
+
+- Ask for help in the issue comments. Maybe you can get back on track and complete the issue.
+
+  - Asking questions is always better than guessing.
+
+- Submit a ``[WIP]`` (work in progress) pull request.
+
+  - If we can see what progress you have made, it is easier to offer help.
+  - Even if you don't complete the project, perhaps someone else can pull in your in-progress work and build on it.
+  - Sometimes your in-progress work is an improvement over not having it, and so we'll merge in something even if it isn't complete.
+
+- If you really cannot move forward, it is okay to abandon an issue.
+
+.. _abandon-issue:
+
+It is okay to abandon an issue
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Sometimes you simply cannot complete work you have said you were going to complete. This could happen because you don't have the skills or knowledge to complete the work, or because the issue cannot actually be completed as scoped, or because you don't have the time.
```

Comment:
don't have all the required skills or knowledge sounds better :P

Reply:
> because you don't have all the required skills or knowledge to complete the work

or

> because you don't have the required skills or knowledge,

?

And why do you think so?

Reply:
A person might have few of the required skills but maybe not related to a specific topic like Android (considering this is documentation)